### PR TITLE
[GBP no update] PDAs only hold one pen at a time

### LINKED ISSUE #19515: Penlight disappears in PDA
**Issue Description**:

my PDA ate my penlight!

**What did you expect to happen**: 

To get both a pen and a penlight from my PDA.

**What happened instead**:

I only got my pen back.

**Why is this bad/What are the consequences**:

Mah Penlight! (its probably not THAT important, it's just eating my penlights)

**Steps to reproduce the problem**:

take pen from PDA
insert penlight to pda
insert pen into PDA
remove pen from PDA
cant remove penlight

**When did the problem start happening**:

Probably since the penlight being able to put into PDA merge

**Extra information**:

NA

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
Tweaks the PDAs so they have a proper pen slot, and work with that instead of trying to locate a pen in their whole contents.

Clipboards already handle pens that way.

## Why It's Good For The Game
Fixes #19515

## Images of changes
Penlight

## Testing
Spawned normally, tried inserting a penlight into a PDA, removed the roundstart pen, inserted the penlight, tried inserting the pen.

Also shot at it to make it sure it deleted properly.

## Changelog
:cl:
fix: PDAs only hold one pen and will not eat up your penlights.
/:cl:

### CODE REVIEW

File: code/modules/pda/PDA.dm [hunk 1]
```diff
@@ -60,6 +60,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 	var/ownrank = null // this one is rank, never alt title
 
 	var/obj/item/paicard/pai = null	// A slot for a personal AI device
+	// The slot where you can store a pen
+	var/obj/item/held_pen
```
Comment (Sirryan2002) [inline code/modules/pda/PDA.dm:64]: You need to account for GC issues here, if held_pen gets deleted you need to clear the reference

Comment (Vi3trice) [reply code/modules/pda/PDA.dm:64]: I feel like I'm missing something. I thought I already accounted for it when removing the pen or when the pda gets destroyed.

Comment (Sirryan2002) [reply code/modules/pda/PDA.dm:64]: if the pen gets QDEL'd the PDA won't know that it doesn't exist anymore and will still hold a reference to it.

You could check to see how books handle items stored inside them and ensures GC

**Review by Sirryan2002 — COMMENTED** [review]
once merge conflict is handled, this looks good code wise

File: code/modules/pda/PDA.dm [hunk 2]
```diff
@@ -214,16 +217,12 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			playsound(src, 'sound/machines/pda_button2.ogg', 50, TRUE)
-			if(ismob(loc))
-				var/mob/M = loc
-				if(M.get_active_hand() == null)
-					M.put_in_hands(O)
-					return
-			O.forceMove(get_turf(src))
+			user.put_in_hands(held_pen)
+			UnregisterSignal(held_pen, COMSIG_PARENT_QDELETING)
+			held_pen = null
```
Comment (farie82) [inline code/modules/pda/PDA.dm:225]: Might want to move this to `clear_pen` and just call that instead.

File: code/modules/pda/PDA.dm [hunk 3]
```diff
@@ -295,18 +294,22 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 	else if(is_pen(C))
-		var/obj/item/pen/O = locate() in src
-		if(O)
+		if(held_pen)
 			to_chat(user, "<span class='notice'>There is already a pen in \the [src].</span>")
 		else
 			user.drop_item()
 			C.forceMove(src)
+			held_pen = C
+			RegisterSignal(held_pen, COMSIG_PARENT_QDELETING, .proc/clear_pen)
```
Comment (farie82) [inline code/modules/pda/PDA.dm:303]: Prob best to make a `add_pen(obj/item/pen/P)` proc which does this. Then make the init also use this

**Review by farie82 — COMMENTED** [review]
Two small things which ensure you only have one register and unregister call in your code

**Review by farie82 — APPROVED** [review]

**Pull request merged**